modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc

```diff
@@ -0,0 +1,131 @@
+* Poll timeout time unit
+* Acknowledgment mode
+* Number of parallel consumers
+|===
+
+[[new_operations]]
+== New Operations
+
+[%header%autowidth.spread]
+|===
+|Kafka Operation | Description | Parameters
+
+| Commit
+a| Commits the offsets associated to a message or batch of messages consumed in a message listener. a| Consumer commit key. The consumer commit key to use to commit the message.
+| Consume | This operation enables receiving messages from one or more Kafka topics. It works very similarly to the Message Listener source, so all the operations that apply to the Message Listener source also apply to this operation. a| * Consumption timeout
+* Timeout time unit
+
+|Describe | Retrieves the subscriptions and assignments of the associated connection, along with the current offsets for each topic-partition. This operation blocks all the consumers, so there are no changes to the offsets assignments or other information that is retrieved by this operation. |
+
+| Publish | The `publish` operation enables sending messages to a Kafka topic, and is non-blocking. Its behavior depends on specific values of the connection parameters. This operation supports transactions, which means that a transaction ID is randomly generated for the Kafka producer and handled in the connection.
+a| *Topic (The topic name)
+* Partition (The partition name)
+* Key
+* Message
+* Headers
+| Seek | Sets the current offset of the consumer for the given topic and partition to the provided offset value. a| * Topic (The topic name)
+* Partition (The partition name)
+* Offset (The offset to seek to) |
+|===
+
+[[new_sources]]
+== New Sources
+
+[%header%autowidth.spread]
+|===
+|Kafka Operation | Description | Parameters
+
+| Batch message listener
+a| Works in the same way as the message listener except that it processes the list of messages instead of processing a single message at a time. The message list that was obtained in the poll is handled by a flow as a single event, so the handling of concurrency is simpler than in the simple message listener. In other words, a commit of the messages happens for all the messages as a whole by calling the commit on Kafka consumer.
+a| * Poll timeout
+* Poll timeout time unit
+* Acknowledgment mode
+* Number of parallel consumers
+|===
+
+
+== Upgrade Prerequisites
+
+Before you perform the upgrade, you must create a backup of your files, data, and configurations in case you need to revert to the previous version.
+
+== Upgrade Steps
+
+Follow these steps to perform the upgrade to Kafka Connector 4.0.x:
+
+. In Studio, create a Mule project.
+. In the Mule Palette view, click *Search in Exchange*.
+. In Add Modules to Project, type 'Kafka' in the search field.
+. In Available modules, select *Kafka Connector*, and then click *Add*.
+. Click *Finish*. +
+Anypoint Studio upgrades the connector automatically.
+. Verify that the `kafka-connector` dependency version is `4.0.x` in the pom.xml.
```

Comment:
This sentence is formatted differently throughout this topic set: please search for and standardize.